### ISSUE TITLE
The probability calculation is different from the others. Why?

### ISSUE THREAD

**Issue by sclang168** [lstich/Baccarat#1]
Dear Mr. / Ms, 
Why do your calculations differ from others? 
Others are: 
    - Player wins with ~= 0.446348 probability
    - Banker wins with ~= 0.458617 probability
    - Tie occurs with  ~= 0.095035
Yours are:
    - Player wins with ~= 0.446147 probability
    - Banker wins with ~= 0.458428 probability
    - Tie occurs with  ~= 0.095426

**Comment by lstich32** [comment]
Hi,
Quickly looking over my code my best guess is because I rounded the hard coded value SINGLE_CARD_PROBABILITY to 9 decimal places.  I probably shouldn’t have since 1/13 is a rational number, if you change this value to 0.07692307692 that might make up the difference

**Comment by sclang168** [comment]
Hi,

&nbsp; &nbsp; I feel that this probability should not be a fixed value, but should change with the number of cards taken. Take one card, the total number is reduced by one, and the number of this card class is also reduced by one, so its probability also changes.

**Comment by lstich32** [comment]
you are correct, however this was a school project and the requirements was with the assumption that the deck has unlimited cards in it ie/ replacements so the probability remains 1/13. This simplifies the calculations and would be causing this discrepancy in the probabilities you’re seeing.

**Comment by sclang168** [comment]
Can you give an accurate probability calculation method for 8 sets of cards?(8*52)

**Comment by lstich32** [comment]
this code could definitely be altered to do that, I dont have time right now to do so but you could pull my code and do so

**Comment by sclang168** [comment]
What instant messaging tool do you use? Would you like to tell me how you joined? For Future Communication and consultation.

**Comment by lstich32** [comment]
sure you can message me on fb messenger you can search me Luke Stichhaller
